src/Kriging.cpp

```diff
@@ -3,876 +3,790 @@
-	Kriging::Vector solve(const Kriging::Vector& b)
-	{
-		Kriging::Vector y(b.size());
-
-		// First solve lower triangular * y = b with forward substitution
-		for (size_t i = 0; i < b.size(); ++i)
-		{
-			double sum = 0.0;
-
-			for (size_t j = 0; j < i; ++j)
-			{
-				sum += (m_matrix[i][j] * y[j]);
-			}
-
-			y[i] = (b[i] - sum) / m_matrix[i][i];
-		}
-
-		// Now solve upper triangular (transpose of lower triangular) * x = y with back substitution.
-		// Note that x can be solved in place using the existing y vector.  No need to allocate 
-		// another vector.
-		for (int i = static_cast<int>(b.size()) - 1; i >= 0; i--)
-		{
-			double sum = 0.0;
-
-			for (int j = static_cast<int>(b.size()) - 1; j > i; j--)
-			{
-				sum += (m_matrix[i][j] * y[j]);
-			}
-
-			y[i] = (y[i] - sum) / m_matrix[i][i];
-		}
-
-		return y;
-	}
+  //! Constructor
+  /** \warning Input matrix is decomposed in-place
+   **/
+  CholeskyDecomposition(const Kriging::Matrix &matrix) : m_matrix(matrix) {
+    assert(matrix.size() > 0 && matrix.size() == matrix[0].size());
+  }
+
+  //! Decomposition into triangular matrices
+  bool decompose() {
+    // Enumerate matrix columnwise
+    for (size_t j = 0; j < m_matrix.size(); ++j) {
+      for (size_t i = j; i < m_matrix.size(); ++i) {
+        if (i == j) {
+          double sum = 0.0;
+
+          for (size_t k = 0; k < i; ++k) {
+            sum += m_matrix[i][k] * m_matrix[i][k];
+          }
+
+          if (m_matrix[i][j] - sum <= 0.0) {
+            // Not positive definite matrix
+
+            return false;
+          }
+
+          m_matrix[i][j] = std::sqrt(m_matrix[i][j] - sum);
+        } else {
+          double sum = 0.0;
+
+          for (size_t k = 0; k < j; ++k) {
+            sum += (m_matrix[i][k] * m_matrix[j][k]);
+          }
+
+          m_matrix[i][j] = (1 / m_matrix[j][j]) * (m_matrix[i][j] - sum);
+          m_matrix[j][i] = m_matrix[i][j];
+        }
+      }
+    }
+
+    return true;
+  }
+
+  //! Solve Ax = b. A is the input matrix.
+  Kriging::Vector solve(const Kriging::Vector &b) {
+    Kriging::Vector y(b.size());
+
+    // First solve lower triangular * y = b with forward substitution
+    for (size_t i = 0; i < b.size(); ++i) {
+      double sum = 0.0;
+
+      for (size_t j = 0; j < i; ++j) {
+        sum += (m_matrix[i][j] * y[j]);
+      }
+
+      y[i] = (b[i] - sum) / m_matrix[i][i];
+    }
+
+    // Now solve upper triangular (transpose of lower triangular) * x = y with
+    // back substitution. Note that x can be solved in place using the existing
+    // y vector.  No need to allocate another vector.
+    for (int i = static_cast<int>(b.size()) - 1; i >= 0; i--) {
+      double sum = 0.0;
+
+      for (int j = static_cast<int>(b.size()) - 1; j > i; j--) {
+        sum += (m_matrix[i][j] * y[j]);
+      }
+
+      y[i] = (y[i] - sum) / m_matrix[i][i];
+    }
+
+    return y;
+  }
 
 protected:
-
-	//! Decomposed matrix
-	Kriging::Matrix m_matrix;
+  //! Decomposed matrix
```

Comment:
And has the spacing changed as well? (should be tabs)

Reply:
it should be a bit more readable now!